inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/random_uniform/random_uniform_kernel_ref.cpp

```diff
@@ -0,0 +1,116 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "random_uniform_kernel_ref.h"
+
+#include <kernel_selector_utils.h>
+
+namespace {
+
+using namespace kernel_selector;
+
+
+size_t GetFullShapeSize(const Tensor::DataTensor &tensor) {
+    return tensor.Batch().v * tensor.Feature().v * tensor.Z().v * tensor.Y().v * tensor.X().v * tensor.W().v;
+}
+
+int getStep(const random_uniform_params &params) {
+    return BytesPerElement(params.output.GetDType()) > 4 ? 2 : 4;
+}
+
+size_t GetGwsSize(const random_uniform_params &params) {
+    size_t shapeSize = GetFullShapeSize(params.output);
+    int step = getStep(params);
+    return (shapeSize >> (step >> 1)) + static_cast<int>(shapeSize % step > 0);
+}
+
+CommonDispatchData SetDefault(const random_uniform_params &params, const optional_params &) {
+    CommonDispatchData dispatchData;
+    dispatchData.gws = {GetGwsSize(params), 1, 1};
+    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+    return dispatchData;
+}
+
+}  // namespace
+
+namespace kernel_selector {
+
+JitConstants RandomUniformKernelRef::GetJitConstants(const random_uniform_params &params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+    // When both seed values are equal to zero RandomUniform should generate non-deterministic sequence.
+    if (params.global_seed == 0 && params.op_seed == 0) {
+        std::srand(std::time(nullptr));
```

Comment:
first, `srand` initializes sequence returned by `rand` and it should only be called once, then all `rand` calls will return pseudo random numbers. second, if you call this function during one second, it will reinitialize `rand` sequence to the same value, which will then generate same seed, breaking requirement, therefore it must not be called second time. this could be accomplished by something like
```c++
static auto initialized = (std::srand(std::time(nullptr)), true);
```

Reply:
as I mentioned already it's about kernel initialization and more like corner case to init the attribute, not random sequence generation. Moreover, the same situation is in reference imlp also. So don't see big deal here. But ok, will add it statically  :)

Reply:
it is kernel initialization, but it's called for every node of this type in every network. you can have several nodes and they will get identical sequences when initialized during one second. i have no idea how much it matters though.
comment in reference impl explicitly allows different implementation in this case.
i should've checked my suggestion with warnings on. it should be
```c++
static auto initialized[[gnu::unused]] = (std::srand(std::time(nullptr)), true);
```